projects/noloan/common/src/main/java/hasadna/noloan/common/FirebaseAuthontication.java

```diff
@@ -0,0 +1,43 @@
+package hasadna.noloan.common;
+
+import android.util.Log;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+import java.util.List;
+
+public class FirebaseAuthontication {
+  private static final String TAG = "FirebaseAuthontication";
+
+  private static FirebaseAuthontication instance;
+  private FirebaseUser user;
+  private FirebaseAuth auth;
+
+  public static FirebaseAuthontication getInstance() {
+    if (instance == null) {
+      instance = new FirebaseAuthontication();
+    }
+    return instance;
+  }
+
+  public void signinAnonymusly() {
+    auth = FirebaseAuth.getInstance();
+    auth.signInAnonymously().addOnCompleteListener(task ->
+    {
+      if (task.isSuccessful()) {
+        user = auth.getCurrentUser();
+      } else {
+        Log.w(TAG, "signInAnonymously:failure", task.getException());
+      }
+    });
+  }
+
+  public boolean containCurrentUserId(List<String> ids) {
+    return ids.contains(user.getUid());
+  }
```

Comment:
This function checks if a user is a suggester of a spam message.

I don't see how it concerns authentication, therefore it doesn't belong to `FirebaseAuthentication`.

You can just check it via the `SmsMessage`:
`SmsMessage.getSuggestersList().contains(FirebaseAuthontication.getInstance().getCurrentUserId());`

OR, if you still think this should be in an external function, have it under `SmsMessages`.

Reply:
Don't care.. gone